File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/GoalItem.kt
package com.goalmaker.app.application.planning

import java.time.LocalDate

/**
 * A goal as the screens and rules see it (docs/goals.md). [mode] is done, tasks or number; [status]
 * is open, done or dropped; [target] and [unit] belong to a numeric goal.
 */
data class GoalItem(
    val id: String,
    val title: String,
    val horizon: GoalHorizon,
    val periodStart: LocalDate,
    val mode: String = GoalRules.MODE_DONE,
    val status: String = GoalRules.OPEN,
    val emoji: String? = null,
    val parentId: String? = null,
    val target: Double? = null,
    val unit: String? = null,
    val completedAt: String? = null,
    val position: Double = 0.0,
    val deleted: Boolean = false,
)

File: windows/src/GoalMaker.Core/Planning/GoalItem.cs
namespace GoalMaker.Core.Planning;

/// <summary>
/// A goal as the screens and rules see it (docs/goals.md). <see cref="Mode"/> is done, tasks or number;
/// <see cref="Status"/> is open, done or dropped; <see cref="Target"/> and <see cref="Unit"/> belong to a numeric goal.
/// </summary>
public sealed record GoalItem(string Id, string Title, GoalHorizon Horizon, DateOnly PeriodStart)
{
    public string Mode { get; init; } = GoalRules.ModeDone;

    public string Status { get; init; } = GoalRules.Open;

    public string? Emoji { get; init; }

    public string? ParentId { get; init; }

    public double? Target { get; init; }

    public string? Unit { get; init; }

    public string? CompletedAt { get; init; }

    public double Position { get; init; }

    public bool Deleted { get; init; }
}
